.github/workflows/prepare-release-proposal.yml

```diff
@@ -0,0 +1,100 @@
+name: Prepare release proposal
+
+on:
+  workflow_dispatch:
+    inputs:
+      base-branch:
+        description: 'Release branch'
+        required: true
+        type: choice
+        options:
+        - v4.x
+        - v5.x
+      release-type:
+        description: 'Release type'
+        required: true
+        type: choice
```

Comment:
Eventually, we might want to release a patch version with more minor commits in `master`, don't we?

Reply:
Why would we want to do that? Usually we release either minor or patch depending on what new commits are on the main branch. I don't see why even a hot fix couldn't be a minor if it includes minor changes.

I guess my main argument against having this as an option is that you end up having to go through the commits to figure this out, which is part of the work this would otherwise remove.

Reply:
The rest of the PR LGTM, so this would be the only thing I'd ideally like to see addressed. I'll still approve the PR so that if you prefer to make this improvement as a second step you're not blocked.

Reply:
I'll try to do some tests in other repo to see if this is easily doable, if it is I'll address it in this PR.

Reply:
Changed